src/Cake.Common/Text/TextTransformation.cs

```diff
@@ -64,7 +64,7 @@ public void Save(FilePath path)
             // Render the content to the file.
             var file = _fileSystem.GetFile(path);
             using (var stream = file.OpenWrite())
-            using (var writer = new StreamWriter(stream, Encoding.UTF8, 1024, true))
+            using (var writer = new StreamWriter(stream, new UTF8Encoding(false), 1024, true))
```

Comment:
So the false parameter here just tells the StreamWriter to preserve BOM (if any) and not add one if none already exist in the stream?